modules/revcontentBidAdapter.js

```diff
@@ -0,0 +1,290 @@
+// jshint esversion: 6, es3: false, node: true
+'use strict';
+
+import {registerBidder} from '../src/adapters/bidderFactory';
+import * as utils from '../src/utils';
+import {ajax} from '../src/ajax';
+
+const BIDDER_CODE = 'revcontent';
+const NATIVE_PARAMS = {
+  title: {
+    id: 0,
+    name: 'title'
+  },
+  image: {
+    id: 3,
+    type: 3,
+    name: 'img'
+  },
+  sponsoredBy: {
+    id: 5,
+    name: 'data',
+    type: 1
+  }
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['banner', 'native', 'video'],
+  isBidRequestValid: function (bid) {
+    return (typeof bid.params.apiKey !== 'undefined' && typeof bid.params.userId !== 'undefined' && bid.hasOwnProperty('nativeParams'));
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const userId = validBidRequests[0].params.userId;
+    const widgetId = validBidRequests[0].params.widgetId;
+    const apiKey = validBidRequests[0].params.apiKey;
+    var domain = validBidRequests[0].params.domain;
+    var host = validBidRequests[0].params.endpoint;
+
+    if (typeof host === 'undefined') {
+      host = 'trends.revcontent.com';
+    }
+
+    let serverRequests = [];
+    var refererInfo;
+    if (bidderRequest && bidderRequest.refererInfo) {
+      refererInfo = bidderRequest.refererInfo.referer;
+    }
+
+    if (typeof domain === 'undefined') {
+      domain = extractHostname(refererInfo);
+    }
+
+    var endpoint = '//' + host + '/rtb?apiKey=' + apiKey + '&userId=' + userId;
+
+    if (!isNaN(widgetId) && widgetId > 0) {
+      endpoint = endpoint + '&widgetId=' + widgetId;
+    }
+
+    let secure = location.protocol === 'https:';
+
+    const imp = validBidRequests.map((bid, id) => {
+      if (bid.hasOwnProperty('nativeParams')) {
+        const assets = utils._map(bid.nativeParams, (bidParams, key) => {
+          const props = NATIVE_PARAMS[key];
+          const asset = {
+            required: bidParams.required & 1
+          };
+          if (props) {
+            asset.id = props.id;
+            let wmin, hmin, w, h;
+            let aRatios = bidParams.aspect_ratios;
+
+            if (aRatios && aRatios[0]) {
+              aRatios = aRatios[0];
+              wmin = aRatios.min_width || 0;
+              hmin = aRatios.ratio_height * wmin / aRatios.ratio_width | 0;
+            }
+
+            asset[props.name] = {
+              len: bidParams.len,
+              type: props.type,
+              wmin,
+              hmin,
+              w,
+              h
+            };
+
+            return asset;
+          }
+        }).filter(Boolean);
```

Comment:
nice usage of `filter(Boolean)`

Reply:
To be fair, I copied that from the openRtb adapter implementation (since we are also using open rtb on the backend as well).